backend/business-partner-agent/src/main/resources/databasemigrations/V1.19__add_chat_messages-table.sql

```diff
@@ -0,0 +1,7 @@
+CREATE TABLE chat_message (
+    id uuid PRIMARY KEY,
+    partner_id uuid NOT NULL,
+    content text,
+    incoming boolean NOT NULL,
+    created_at timestamp without time zone
+);
```

Comment:
Partner FK constrain is missing as it should not be possible to delete a partner that still has chat messages. Or if so all messages should be deleted when the partner is. So either restrict or cascade here.

Reply:
Are we using FK now? I thought we didn't use FK.  I am confused (again).

Reply:
just stuck with explicit call to delete when partner removed

Reply:
My original line of thought was that we can get away without constraints, by keeping entities simple buckets with minimal relations between each other. If additionally all calls go through managers the manager can then decide if related entities need to be updated or deleted. But as we started to add more and more joins and relations this does not work any more. We had huge issues with the partner - schema - credential relations. Because deleting a partner did not update the partner id in the credential (exchange) so the UI was broken. Another issue was the default schema loading on startup (see https://github.com/hyperledger-labs/business-partner-agent/pull/520). 

With that many relations between entities we need to change our approach, because otherwise the risk of broken data is too great. So each relation needs to be backed up by a foreign key constraint in the DB. The default of restrict is what we want in most cases. For the relation partner -> credential exchange I added a cascade delete though.

Reply:
OK, sounds good.  Activity and Chat Messages are trivial data, so I will just add a migration to have FK and cascade delete.